### PR TITLE
Boss 3

### PR DESCRIPTION
**Pull request by JaceCear**

### CODE REVIEW

File: include/sprite.h [hunk 1]
```diff
@@ -173,6 +172,19 @@ typedef struct {
     /* 0x28 */ Hitbox hitboxes[1];
 } Sprite /* size = 0x30 */;
 
+// TODO: Unify Sprite with variable hitbox count through a macro
```
Comment (freshollie) [inline include/sprite.h:175]: So this would be like `SPRITE(2)` or something?

File: src/game/bosses/boss_3.c [hunk 2]
```diff
@@ -0,0 +1,1811 @@
...
+    totem->unk33 = 240;
+    totem->unk30 = 0;
+    totem->unk34 = 0;
+    totem->unk35 = 0;
+    totem->unk36 = 0;
+
+    totem->unkB4 = 120;
+    totem->tilesBullet = VramMalloc((2 * 2) * 2);
+
+    s = &totem->sprC0.s;
```
Comment (freshollie) [inline src/game/bosses/boss_3.c:322]: Do none of the init macros work for these?

Comment (freshollie) [reply src/game/bosses/boss_3.c:322]: Or because it's non matching

File: include/sprite.h (hunk 1 above)
Comment (JaceCear) [reply include/sprite.h:175]: Yeah, something like that, but make it two:
One for definition, one for declaration.

File: src/game/bosses/boss_3.c (hunk 2 above)
Comment (JaceCear) [reply src/game/bosses/boss_3.c:322]: Mostly because it's nonmatching

**Review by freshollie — APPROVED** [review]
Good to merge. The stack non matches I might be able to find a solution with the permuter but can merge first :)

**Pull request merged**